From 2c0bf64129dfdb96e3b3748f52f9a6eed7e0268f Mon Sep 17 00:00:00 2001
From: Christine Kim <125395064+chkim-usgs@users.noreply.github.com>
Date: Tue, 23 Sep 2025 16:21:22 -0400
Subject: [PATCH] Using updated cspice (#103)

* working with new cspice

* update changelog

* Update CMakeLists.txt

* fix tests
---
 CHANGELOG.md                 |  1 +
 CMakeLists.txt               |  9 ++++++---
 SpiceQL/tests/CMakeLists.txt |  7 ++++---
 cmake/FindCSpice.cmake       | 21 ---------------------
 4 files changed, 11 insertions(+), 27 deletions(-)
 delete mode 100644 cmake/FindCSpice.cmake

diff --git a/CHANGELOG.md b/CHANGELOG.md
index eef3a3a..4aac20c 100644
--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -40,6 +40,7 @@ release.
 ### Changed
 - Refactored FastAPI models and added OpenAPI examples [#96](https://github.com/DOI-USGS/SpiceQL/pull/96)
 - Pinned cspice to 67 [#102](https://github.com/DOI-USGS/SpiceQL/pull/102)
+- Update cmake files to accommodate new cspice package that uses the new cmake recipe [#103](https://github.com/DOI-USGS/SpiceQL/pull/103)
 
 ### Added
 - Added relative kernel path support for furnishing [#100](https://github.com/DOI-USGS/SpiceQL/pull/100)
diff --git a/CMakeLists.txt b/CMakeLists.txt
index fae8746..c802856 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -55,7 +55,7 @@ if(SPICEQL_BUILD_LIB)
   add_subdirectory("submodules/gularkfilesystem")
   add_subdirectory("submodules/json")
   
-  find_package(CSpice REQUIRED)
+  find_package(cspice REQUIRED)
   find_package(fmt REQUIRED)
   find_package(cereal REQUIRED)
   find_package(spdlog REQUIRED)
@@ -136,7 +136,9 @@ if(SPICEQL_BUILD_LIB)
 
   target_compile_definitions(SpiceQL PRIVATE SPDLOG_ACTIVE_LEVEL=SPDLOG_LEVEL_TRACE 
                                      PUBLIC -D_SOURCE_PREFIX="${CMAKE_CURRENT_SOURCE_DIR}")
-  
+
+get_target_property(CSPICE_INCLUDE_DIRS CSPICE::cspice INTERFACE_INCLUDE_DIRECTORIES)
+
 target_include_directories(SpiceQL
                             PUBLIC
                             $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/SpiceQL/include>
@@ -144,6 +146,7 @@ target_include_directories(SpiceQL
                             PRIVATE
                             ${CMAKE_CURRENT_SOURCE_DIR}/submodules/BTree/include/
                             cereal
+                            ${CSPICE_INCLUDE_DIRS}
                             )
 
   target_link_libraries(SpiceQL
@@ -151,7 +154,7 @@ target_include_directories(SpiceQL
                         fmt::fmt-header-only
                         nlohmann_json::nlohmann_json
                         PRIVATE
-                        cspice
+                        CSPICE::cspice
                         spdlog::spdlog_header_only
                         HighFive
                         ${CURL_LIBRARIES}
diff --git a/SpiceQL/tests/CMakeLists.txt b/SpiceQL/tests/CMakeLists.txt
index a4d3624..b4ece43 100644
--- a/SpiceQL/tests/CMakeLists.txt
+++ b/SpiceQL/tests/CMakeLists.txt
@@ -3,7 +3,7 @@ cmake_minimum_required(VERSION 3.10)
 # A lot of linux OSes do not support C++20
 set(CMAKE_CXX_STANDARD 20)
 
-find_package(CSpice REQUIRED)
+find_package(cspice REQUIRED)
 
 get_property(dirs DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY INCLUDE_DIRECTORIES)
 foreach(dir ${dirs})
@@ -30,7 +30,7 @@ target_compile_definitions(runSpiceQLTests PUBLIC SPDLOG_ACTIVE_LEVEL=SPDLOG_LEV
 target_link_libraries(runSpiceQLTests
                       PRIVATE
                       SpiceQL
-                      cspice
+                      CSPICE::cspice
                       gtest
                       gmock
                       Threads::Threads
@@ -40,6 +40,7 @@ target_link_libraries(runSpiceQLTests
 
 
 target_include_directories(runSpiceQLTests PRIVATE ${CMAKE_SOURCE_DIR}/submodules/hippomocks/ 
-                                                   ${CMAKE_SOURCE_DIR}/submodules/BTree/include/)
+                                                   ${CMAKE_SOURCE_DIR}/submodules/BTree/include/
+                                                   ${CSPICE_INCLUDE_DIRS})
 
 gtest_discover_tests(runSpiceQLTests WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/SpiceQL/tests/)
diff --git a/cmake/FindCSpice.cmake b/cmake/FindCSpice.cmake
deleted file mode 100644
index b1d9965..0000000
--- a/cmake/FindCSpice.cmake
+++ /dev/null
@@ -1,21 +0,0 @@
-# CMake module for find_package(CSPICE)
-# Finds include directory and all applicable libraries
-#
-# Sets the following:
-#   CSPICE_INCLUDE_DIR
-#   CSPICE_LIBRARY
-
-find_path(CSPICE_INCLUDE_DIR
-  NAME SpiceUsr.h
-  PATH_SUFFIXES naif cspice
-)
-
-find_library(CSPICE_LIBRARY
-  NAMES cspice
-)
-
-link_directories($ENV{CONDA_PREFIX}/lib)
-include_directories(${CSPICE_INCLUDE_DIR})
-
-message(STATUS "CSPICE INCLUDE: " ${CSPICE_INCLUDE_DIR} )
-message(STATUS "CSPICE LIB: "  ${CSPICE_LIBRARY} )
-- 
2.39.5 (Apple Git-154)

